### DISCUSSION TITLE
Text is erased on save with garbage collection turned on

### DISCUSSION THREAD

**Discussion by mwade625** [pymupdf/PyMuPDF#2231] [Q&A]
In some circumstances I am seeing text being removed from the page when I have applied redactions and then saved with the garbage parameter set to anything > 0 on the save.  Has anyone run into this kind of behavior?

Thanks,

Mike

**Comment by JorjMcKie** [comment]
Well, redacted text **_should_** be removed. That's the purpose in the life of a redaction.
But presumably you are encountering text removals, that had not been covered by a redact rectangle before, right?
If so, then you probably have seen a duplicate of issue #2108.

**Comment by mwade625** [comment]
This was a very poor description of the problem, my apologies.  I am using the latest version 1.21.1  of pymupdf.  

The issue that I am seeing may be related.  I saw lots of issues when trying incremental saves, so I don't do them that at all.  I just save to a new file.  However, recently I saw an issue when saving that if I had the garbage value set to '4' when apply redactions, occasionally most of the text (un-redacted text) would just disappear from some of the pages.  I reduced this value to '3' and things were much better.  The other day I was doing some testing and noted that even with garbage set to 3 on the save I would occasionally see this happen.  In this example I had a paragraph and when I applied the redaction to one word the rest of the paragraph (multiple lines) would come up blank.  

When I set garbage to 0 on the save then this doesn't happen.  I don't have many examples where I caught this happening and unfortunately the one reliable example I have I can't share with you.  

When do you expect the next release to come out?  And was that bug you reported #2108 specific to this version?  (Just wondering if I could roll back to an earlier release.

Thanks

**Comment by JorjMcKie** [comment]
Well, there is no other choice than attacking issues one by one, is there?

When you say, you don't use incremental saves because of "lots of issues", then I cannot simply look away and let you just no longer use them. **_Please be specific with a description and provide an example!_**

In my locally installed pre-version, I can try things. You can send examples to my e-mail so confidential stuff will not become public.

**Comment by mwade625** [comment]
I can't do it today, but I will step thru some PDF's that I have and try to recreate the problems so I can give you something to work with. My reports here were not very helpful I know.
